### ISSUE TITLE
the website is empty sometimes

### ISSUE THREAD

**Issue by pipi815** [jinseob2kim/jsmodule#25]
the website is empty sometimes，how to deal with.

**Comment by jinseob2kim** [comment]
Hello. please share the code or screenshot?

**Comment by pipi815** [comment]
<img width="1391" alt="image" src="https://github.com/jinseob2kim/jsmodule/assets/147511301/24cc9793-a885-4946-b1e3-4fd045b8c4fb">
> jsBasicGadget(xinzang)

Listening on http://127.0.0.1:7571

**Comment by jinseob2kim** [comment]
Is "xinzang" data.frame?

**Comment by pipi815** [comment]
Yes and I've specified all the variables in the data frame, so it doesn't happen all the time but it happens sometimes

**Comment by jinseob2kim** [comment]
can you show `str(xinzang)` or `summary(xinzang)` ? I want to check the data

**Comment by pipi815** [comment]
Yes, of course. Please wait a minute. I used the same data box to load it successfully once this morning, but it has not been loaded again for some reason. It turns out that it will often load sometimes it can load sometimes it can't load I don't know why.

**Comment by pipi815** [comment]
> str(xinzang)
tibble [319,795 × 17] (S3: tbl_df/tbl/data.frame)
 $ xinzang     : Factor w/ 2 levels "是","否": 1 1 1 1 1 2 1 1 1 1 ...
 $ BMI         : num [1:319795] 16.6 20.3 26.6 24.2 23.7 ...
 $ xiyan       : Factor w/ 2 levels "是","否": 2 1 2 1 1 2 1 2 1 1 ...
 $ yinjiu      : Factor w/ 2 levels "是","否": 1 1 1 1 1 1 1 1 1 1 ...
 $ zhongfeng   : Factor w/ 2 levels "是","否": 1 2 1 1 1 1 1 1 1 1 ...
 $ shenti      : num [1:319795] 3 0 20 0 28 6 15 5 0 0 ...
 $ xinli       : num [1:319795] 30 0 30 0 0 0 0 0 0 0 ...
 $ buxing      : Factor w/ 2 levels "是","否": 1 1 1 1 2 2 1 2 1 2 ...
 $ xingbie     : Factor w/ 2 levels "女","男": 1 1 2 1 1 1 1 1 1 2 ...
 $ nianling    : Factor w/ 13 levels "18-24","25-29",..: 8 13 10 12 5 12 11 13 13 10 ...
 $ renzhong    : Factor w/ 6 levels "印第安人","亚洲人",..: 6 6 6 6 6 3 6 6 6 6 ...
 $ tangniaobing: Factor w/ 2 levels "是","否": 2 1 2 1 1 1 1 2 1 1 ...
 $ huodong     : Factor w/ 2 levels "是","否": NA NA NA NA NA NA NA NA NA NA ...
 $ shuimian    : num [1:319795] 5 7 8 6 8 12 4 9 5 10 ...
 $ xiaochuan   : Factor w/ 2 levels "是","否": 2 1 2 1 1 1 2 2 1 1 ...
 $ shenzang    : Factor w/ 2 levels "是","否": 1 1 1 1 1 1 1 1 2 1 ...
 $ pifu        : Factor w/ 2 levels "是","否": 2 1 1 2 1 1 2 1 1 1 ...
> summary(xinzang)
 xinzang          BMI        xiyan       yinjiu      zhongfeng       shenti           xinli        buxing     
 是:292422   Min.   :12.02   是:187887   是:298018   是:307726   Min.   : 0.000   Min.   : 0.000   是:275385  
 否: 27373   1st Qu.:24.03   否:131908   否: 21777   否: 12069   1st Qu.: 0.000   1st Qu.: 0.000   否: 44410  
             Median :27.34                                       Median : 0.000   Median : 0.000              
             Mean   :28.33                                       Mean   : 3.372   Mean   : 3.898              
             3rd Qu.:31.42                                       3rd Qu.: 2.000   3rd Qu.: 3.000              
             Max.   :94.85                                       Max.   :30.000   Max.   :30.000              
                                                                                                              
 xingbie        nianling          renzhong      tangniaobing huodong          shuimian      xiaochuan   shenzang   
 女:167805   65-69  : 34151   印第安人:  5202   是:276434    是  :     0   Min.   : 1.000   是:276923   是:308016  
 男:151990   60-64  : 33686   亚洲人  :  8068   否: 43361    否  :     0   1st Qu.: 6.000   否: 42872   否: 11779  
             70-74  : 31065   黑人    : 22939                NA's:319795   Median : 7.000                          
             55-59  : 29757   西班牙裔: 27446                              Mean   : 7.097                          
             50-54  : 25382   其他人种: 10928                              3rd Qu.: 8.000                          
             ≥80    : 24153   白人    :245212                              Max.   :24.000                          
             (Other):141601                                                                                        
 pifu       
 是:289976  
 否: 29819

**Comment by jinseob2kim** [comment]
Thanks. We're not sure why yet, but we'll keep checking and let you know. Thanks for letting us know!

**Comment by pipi815** [comment]
Thanks for your reply. Is there any other way I can use this feature that I'm anxious to use right now

**Comment by jinseob2kim** [comment]
You can run `jsBasicExtAddin()` and upload your data file.

**Comment by pipi815** [comment]
Yes, I have also used this method, but I still reported an error and successfully ran it once in the morning, and then I couldn't run it again. Besides, I have cleared all the use environment and restarted the computer, which didn't get a good effect. Besides, there is no way to define the factor variables. Thank you very much.
> jsBasicExtAddin(xinzang)

Listening on http://127.0.0.1:7571
Warning in Ops.factor(left, right) : ‘<=’ not meaningful for factors
Warning in Ops.factor(left, right) : ‘<=’ not meaningful for factors
Warning in Ops.factor(left, right) : ‘<=’ not meaningful for factors
Warning in Ops.factor(left, right) : ‘<=’ not meaningful for factors
Warning in Ops.factor(left, right) : ‘<=’ not meaningful for factors
Warning in Ops.factor(left, right) : ‘<=’ not meaningful for factors
Warning in Ops.factor(left, right) : ‘<=’ not meaningful for factors
Warning in Ops.factor(left, right) : ‘<=’ not meaningful for factors
Warning in Ops.factor(left, right) : ‘<=’ not meaningful for factors
Warning in Ops.factor(left, right) : ‘<=’ not meaningful for factors
Warning in Ops.factor(left, right) : ‘<=’ not meaningful for factors
Warning in Ops.factor(left, right) : ‘<=’ not meaningful for factors
Warning in Ops.factor(left, right) : ‘<=’ not meaningful for factors
Warning: The select input "datafile-factor_vname" contains a large number of options; consider using server-side selectize for massively improved performance. See the Details section of the ?selectizeInput help topic.
File 心脏病数据.csv was uploaded
<img width="1420" alt="image" src="https://github.com/jinseob2kim/jsmodule/assets/147511301/bab25155-dfa5-401c-a921-2f9063ee2fe4">
<img width="1304" alt="image" src="https://github.com/jinseob2kim/jsmodule/assets/147511301/0f8fa654-e901-4b00-b854-c9d37b044c41">
